### PR TITLE
Translate Program Comment Section

### PR DESCRIPTION
**Pull request by JoshuaSchaefer**
---
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [ ] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no warnings and errors 
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Verify that all tests are passing (CI), if not please state the test cases in the [section](#Tests) below
- [x] Perform a self-review of the changes
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [x] Post a message in the *#catroweb* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer
- [x] Check that your pull request has been successfully deployed to https://web-test-1.catrob.at/

### Additional Description
The program comment section has been updated to utilize the new iTranslate API and languages API to translate user-generated content. Now, when the translate button is selected, a new translation section will appear with the original content translated to the user's preferred language using the iTranslate API. Within the new translation section, there is also a statement which displays the translation provider, the source language, and the target language. There is also logic to handle different orderings of the provider, source language, and target language to support various languages with different grammars.

Additional information on the UI design can be found in the original mockup document through the following link.
https://docs.google.com/document/d/1ELqaGX_zIUMP5p8bcGcGhIe8gDj55ITANxYYuFRgePQ/edit?usp=sharing

### Tests - additional information
Updated the Behat translation tests for comment sections to select the translation button and check if the new translation section opens as intended. These new tests replace the previous tests which checked if the href attribute was updated properly.

### CODE REVIEW

File: translations/catroweb.en.yml [hunk 1]
```diff
@@ -182,6 +182,9 @@ programs:
   link: Copy project link to clipboard
   reactionsText: Reactions
   downloadErrorText: Error occurred while downloading the project. Please try again later.
+  showTranslation: Show Translation
```
Comment (dmetzner) [inline translations/catroweb.en.yml]: Show translation*
 We try to be grammatically correct with upper/lowercase

File: translations/catroweb.en.yml [hunk 2]
```diff
@@ -182,6 +182,9 @@ programs:
...
+  hideTranslation: Hide Translation
```
Comment (dmetzner) [inline translations/catroweb.en.yml]: Hide translation

**Review by dmetzner — CHANGES_REQUESTED** [review]
Looks good to me. Only the translations have to be "fixed" :)


Edit: Some screens for UX: @ThorstenBandel 
image
image

**Comment by ThorstenBandel** [comment]
Looks good, just one thing: can you User the report icon from google fonts? https://fonts.google.com/icons?selected=Material%20Icons%20Outlined%3Areport

And please dont make the icon red but gray as in the mockups. We dont want to emphatize on the reporting

**Comment by JoshuaSchaefer** [comment]
> Looks good, just one thing: can you User the report icon from google fonts? https://fonts.google.com/icons?selected=Material%20Icons%20Outlined%3Areport
> 
> And please dont make the icon red but gray as in the mockups. We dont want to emphatize on the reporting

I think this work might be out of the scope of this PR. This PR is just to add the ability to translate a comment with the new API and replace the original comment with the new translated comment on the UI. I don't believe there are any changes to the existing report icon or any functionality related to it.

File: translations/catroweb.en.yml (hunk 1 above)
Comment (JoshuaSchaefer) [reply translations/catroweb.en.yml]: Updated to "Show translation"

File: translations/catroweb.en.yml (hunk 2 above)
Comment (JoshuaSchaefer) [reply translations/catroweb.en.yml]: Updated to "Hide translation"

**Review by dmetzner — APPROVED** [review]

**Pull request merged**